### PR TITLE
feat: support .js and JSX in components

### LINKED ISSUE #2352: how to use .js/.jsx file instead .vue for page?
is there way to use jsx in pages with out using .vue file?

it seem ugly to write `<script> </script>` as it usually only have js inside.

### PR DESCRIPTION
**Pull request by clarkdo**
#### Resolve #2352
- [x] resolve .js under layouts and pages
- [x] watch .js under layouts and pages
- [x] support JSX syntax (already supported in `babel-preset-vue-app`)
- [x] add examples
- [x] add tests

### CODE REVIEW

**Review by atinux — APPROVED** [review]

File: .eslintrc.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ module.exports = {
     browser: true,
     node: true
   },
-  extends: 'standard',
+  extends: ["standard", "standard-jsx"],
```
Comment (Atinux) [inline .eslintrc.js]: Could you use `'` instead of `"`?

File: examples/jsx/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "hello-nuxt-jsx",
+  "name": "nuxt-jsx",
   "dependencies": {
-    "nuxt": "latest"
+    "nuxt": "next"
```
Comment (Atinux) [inline examples/jsx/package.json]: I believe we should keep `latest` since we won't have to update examples when 1.0 will be released.

**Review by atinux — COMMENTED** [review]
Great changes, I commented on 2 small things, otherwise it's perfect, like always :dancer:

File: .eslintrc.js (hunk 1 above)
Comment (clarkdo) [reply .eslintrc.js]: oops, my mistake.

File: examples/jsx/package.json (hunk 2 above)
Comment (clarkdo) [reply examples/jsx/package.json]: I just want to make it useable for now and change it when 1.0 is going to release.
But now I think you're right, keep  consistency is more important, I will make the change. 😸

Comment (Atinux) [reply examples/jsx/package.json]: For information, I always try the examples with: `bin/nuxt examples/hello-world/` :)

**Comment by Atinux** [comment]
:raised_hands: :raised_hands: :raised_hands:

**Pull request merged**